viewer/packages/pointclouds/src/PointCloudPickingHandler.ts

```diff
@@ -0,0 +1,102 @@
+/*!
+ * Copyright 2021 Cognite AS
+ */
+
+import * as THREE from 'three';
+
+import { IntersectInput } from '@reveal/model-base';
+import { PointCloudNode } from './PointCloudNode';
+
+import { PickPoint, PointCloudOctree, PointCloudOctreePicker } from './potree-three-loader';
+
+export interface IntersectPointCloudNodeResult {
+  /**
+   * Distance from camera to intersected point.
+   */
+  distance: number;
+  /**
+   * Coordinate of the intersected point.
+   */
+  point: THREE.Vector3;
+  /**
+   * Point index in the point cloud of the intersected point.
+   */
+  pointIndex: number;
+  /**
+   * Point cloud node defining what model the point is a part of.
+   */
+  pointCloudNode: PointCloudNode;
+  /**
+   * The geometry object that was intersected.
+   */
+  object: THREE.Object3D;
+}
+
+export class PointCloudPickingHandler {
+  private readonly _normalized = new THREE.Vector2();
+  private readonly _raycaster = new THREE.Raycaster();
+  private readonly _picker: PointCloudOctreePicker;
+
+  constructor(renderer: THREE.WebGLRenderer) {
+    this._picker = new PointCloudOctreePicker(renderer);
+  }
+
+  intersectPointClouds(
+    nodes: PointCloudNode[],
+    input: IntersectInput,
+    threshold: number = 0.05 // 5 cm
+  ): IntersectPointCloudNodeResult[] {
+    const { normalizedCoords, camera } = input;
+    this._normalized.set(normalizedCoords.x, normalizedCoords.y);
+    this._raycaster.setFromCamera(normalizedCoords, camera);
+    this._raycaster.params.Points = { threshold };
+
+    const intersections: PickPoint[] = [];
+
+    nodes.forEach(node => {
+      const intersection = this._picker.pick(camera, this._raycaster.ray, [node.potreeNode.octree], {
+        pickWindowSize: 20
```

Comment:
What does 20 mean?

Reply:
It's default PickingWindowSize that is a region on the screen around the cursor that will be checked for points there.